### PR TITLE
Create ci.yml

### PR DESCRIPTION
**Pull request by thusson13**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  push:
+    branches:
+      - '*'
+      - '!dev'
+      - '!master' # remove once main branch is dev
+jobs:
+  build-and-test:
+    name: Run Build and UI Tests
+    runs-on: macos-latest
```
Comment (kenyee) [inline .github/workflows/ci.yml:11]: does this really need to be built on a Mac?
They have linux build agents too (mac agents are usually slower to queue)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          java-version: 1.8
+
+      - name: Build
+        working-directory: ./BrickKit
+        run: ./gradlew build
+      
+      - name: Run Tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 29
```
Comment (kenyee) [inline .github/workflows/ci.yml:28]: should this run on minsdk as well?

**Review by kenyee — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thusson13) [reply .github/workflows/ci.yml:11]: The emulator action suggests using macos https://github.com/ReactiveCircus/android-emulator-runner

`It is recommended to run this action on a macOS VM, e.g. macos-latest or macos-10.15 to take advantage of hardware accleration support provided by HAXM.`

File: .github/workflows/ci.yml (hunk 2 above)
Comment (thusson13) [reply .github/workflows/ci.yml]: Trie that but it seems to blow up inside mockito mockmaker. Created https://github.com/wayfair/brickkit-android/issues/147 to add that

**Pull request closed without merging**